client/actions/auth.js

```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
   };
 }
 
+// calls checkSession to refresh idToken
+function refreshToken () {
+  return new Promise((resolve, reject) => {
+    // invoke check session to get a new token
+    webAuth.checkSession({},
+      function(err, result) {
+        if (err) { // there was an error
+          reject(err);
+        } else {  // we got a token
+          resolve(result);
+        }
+      }
+    );
+  });
+}
+
 const processTokens = (dispatch, apiToken, returnTo) => {
-  const decodedToken = jwtDecode(apiToken);
-  if (isExpired(decodedToken)) {
-    return;
-  }
 
-  axios.defaults.headers.common.Authorization = `Bearer ${apiToken}`;
-  axios.defaults.headers.common['dae-locale'] = window.config.LOCALE || 'en';
+ return new Promise((resolve, reject) => {
+    const decodedToken = jwtDecode(apiToken);
+    if (isExpired(decodedToken)) {
+      return;
+    }
 
-  sessionStorage.setItem('delegated-admin:apiToken', apiToken);
+    axios.defaults.headers.common.Authorization = `Bearer ${apiToken}`;
+    axios.defaults.headers.common['dae-locale'] = window.config.LOCALE || 'en';
+
+    sessionStorage.setItem('delegated-admin:apiToken', apiToken);
```

Comment:
we should be storing the expiresAt time here "instead" of the apiToken.  We shouldn't be storing the API token in local or sessionStorage.  Too easy for someone to nab it from there.  I know this was already doing that, but we might want to just fix this now.